18_regexp_lab/01.py

```diff
@@ -11,20 +11,27 @@
 """
 import re
 import argparse
+from collections import defaultdict
+
 
 parser = argparse.ArgumentParser()
 parser.add_argument("file", type=str, help="The file you want to scan")
 parser.add_argument("key", type=str, help="sreach of key value")
 args = parser.parse_args()
 
 
-def getvaule(key, f):
-    #(?<=key = ).*
-    res = re.search(r"(?<=" + key + r' = ).*', f)
-    if res is not None:
-        return res.group(0)
-    else: return "key not found"
 
+dic = defaultdict(list)
 with open(args.file, "r") as f:
-    print getvaule(args.key, f.read())
-    
\ No newline at end of file
+    for line in f:
+        #re.search(r"^\W*number\W*=\W*(.*)', line)
+        res = re.search(r"^\W*" + args.key + r'\W*=\W*(.*)', line)
+        if res is not None:
+            dic[args.key].append(res.group(1))    
+            
+if dic.get(args.key) is not None:
+    for item in dic.get(args.key):
+        print item
+else:
+    print "key not found"
+    
```

Comment:
It would be probably better to first read and parse the whole file into a dictionary and then use that to look up the expected field. I see you prepare for multiple entries with the same key. Nice, though usually that is disregarded in similar config files. The last one wins. (Though I think I'd add a warning message if I find a duplicate key.)

Lastly, instead of `if res is not None:` I'd just write `if res:` and instead of `if dic.get(args.key) is not None:` I'd write `if args.key in dic:` those are more Pythonic. Oh and I'd not use 'dic' as a variable name. That's just confusing.